espressif/vscode-esp-idf-extension
# [VSC-1523 | 1710 | 1567] EIM Integration

## Description

Add initial EIM integration with `eim_idf.json` and command to open URL for IDF-IM-UI download.

## Type of change

- New feature (breaking change which adds functionality)

## Steps to test this pull request

This PR introduces breaking change to how this extension works. 

1) Use `eim_idf.json` to get ESP-IDF setups in the user environment. If no setups were found, fallback to resolve system environment variables IDF_TOOLS_PATH `esp_idf.json` and `idf.customExtraVars` ["IDF_TOOLS_PATH"] `esp_idf.json`. If no setups are found, it will try to use environment variables `IDF_PATH`, `IDF_TOOLS_PATH` and `IDF_PYTHON_ENV_PATH` to define the current IDF_SETUP. These variables can also be defined in `idf.customExtraVars`.
2) Use workspace folder state to define the current IDF Setup for current workspace folder. IDF_PATH, IDF_TOOLS_PATH, IDF_PYTHON_ENV_PATH and other variables are used to have a better mapping with ESP-IDF. These can be see in `ESP-IDF: Doctor Command`.
3) When the user select a IDF Setup from the dropdown, the extension will use the activation script created from EIM ESP-IDF setup if available. 
4) Remove old debug adapter TypeScript code.
5) Add `ESP-IDF: Open ESP-IDF Install manager`command which open a URL to download EIM releases.
6) On extension activation, a setting `"idf.currentSetup"` will be read to load env variables into workspace state of the corresponding esp-idf setup. Example: `"idf.currentSetup": "/path/to/esp-idf"`

Things to test:

1) Start this extension with already configured extension from before. While extension will not recognize current settings, the user can still select from dropdown and list of existing setups should appear. Selecting the new setup should works without issues. **PLEASE HELP WITH IDEAS TO MINIMIZE ISSUES AND MANUAL STEPS HERE**.
2) Pick a ESP-IDF setup from status bar esp-idf version (ESP-IDF: Select Current ESP-IDF Version). This step assume no EIM setup has been done yet. The list of ESP-IDF versions from before (`$IDF_TOOLS_PATH/esp_idf.json`) should be listed. When one esp-idf setup is selected, settings are saved in current workspace folder in workspace state and a setting `"idf.currentSetup"` will be saved pointing to selected esp-idf.
3) Perform a EIM setup using EIM (either UI or CLI). Now when using this extension and click at status bar esp-idf version (ESP-IDF: Select Current ESP-IDF Version), the list of ESP-IDF should also come from EIM's eim_idf.json. When one esp-idf setup is selected, settings are saved in current workspace folder in workspace state and a setting `"idf.currentSetup"` will be saved pointing to selected esp-idf.

## How has this been tested?

Manual testing of scenarios described above.

**Test Configuration**:
* ESP-IDF Version: 5.3.1 and 5.4
* OS (Windows,Linux and macOS): macOS

## Checklist
